server/api.js

```diff
@@ -588,6 +593,19 @@ router.postAsync("/stage", ensure.isPooler, async (req, res) => {
   const tourney = await Tournament.findOne({ code: req.body.tourney });
   tourney.stages[req.body.index].mappack = req.body.stage.mappack;
   tourney.stages[req.body.index].poolVisible = req.body.stage.poolVisible;
+
+  // Only admin is allowed to toggle stats visibility (undefined is treated as false)
+  if (
+    req.body.stage.statsVisible !== undefined &&
+    (req.body.stage.statsVisible != tourney.stages[req.body.index].statsVisible ?? false)
```

Comment:
this condition is hard to understand, since it's unclear if it's intended to be 
```
req.body.stage.statsVisible !== (tourney.stages[req.body.index].statsVisible ?? false)
``` 
or 
```
(req.body.stage.statsVisible !== tourney.stages[req.body.index].statsVisible) ?? false
```
i assumed you meant the former but javascript interprets it as the latter

Reply:
whoops